test/v1alpha1/route.go

```diff
@@ -129,6 +129,12 @@ func IsRouteReady(r *v1alpha1.Route) (bool, error) {
 	return r.Generation == r.Status.ObservedGeneration && r.Status.IsReady(), nil
 }
 
+// IsRouteNotReady will check the status conditions of the route and return true if the route is
+// not ready.
+func IsRouteNotReady(r *v1alpha1.Route) (bool, error) {
+	return !r.Status.IsReady(), nil
```

Comment:
The IsRouteReady check has a generation check that we probably want here

`r.Generation == r.Status.ObservedGeneration`

Reply:
@dprotaso Thank you!

But it seems that `routes.status.ObservedGeneration` is not assigned until the Route becomes ready. So, the test fails due to `Generation=1` vs `ObservedGeneration=0` mismatch.
For example, when we deployed invalid image below,

```
cat <<EOF | kubectl apply -f -
apiVersion: serving.knative.dev/v1alpha1
kind: Service
metadata:
  name: helloworld-go
spec:
  runLatest:
    configuration:
      revisionTemplate:
        spec:
          container:
            image: image-registry.openshift-image-registry.svc:5000/knative-serving/invalidhelloworld
            resources:
              limits:
                cpu: 100m
                memory: 200M
              requests:
                cpu: 100u
                memory: 100M
            env:
              - name: TARGET
                value: "Go Sample v1"
EOF
```

`observedGeneration` is not assigned and does have `0` for `observedGeneration`  until it becomes Ready.

```
$ kubectl get rt -o yaml helloworld-go |grep -E "generation|observedGeneration"
  generation: 1
```